test/Microsoft.Framework.Runtime.FunctionalTests/ProjectFileGlobbing/ProjectFileGlobbingFacts.cs

```diff
@@ -0,0 +1,178 @@
+                @"src\project\..\..\lib\sub3\source7.cs",
+                @"src\project\..\..\lib\sub4\source8.cs",
+                @"src\project\source1.cs");
+        }
+
+        private DisposableProjectContext CreateContext()
+        {
+            var context = new DisposableProjectContext();
+            context.AddFiles(
+                "src/project/source1.cs",
+                "src/project/sub/source2.cs",
+                "src/project/sub/source3.cs",
+                "src/project/sub2/source4.cs",
+                "src/project/sub2/source5.cs",
+                "src/project/compiler/preprocess/preprocess-source1.cs",
+                "src/project/compiler/preprocess/sub/preprocess-source2.cs",
+                "src/project/compiler/preprocess/sub/sub/preprocess-source3.cs",
+                "src/project/compiler/preprocess/sub/sub/preprocess-source3.txt",
+                "src/project/compiler/shared/shared1.cs",
+                "src/project/compiler/shared/shared1.txt",
+                "src/project/compiler/shared/sub/shared2.cs",
+                "src/project/compiler/shared/sub/shared2.txt",
+                "src/project/compiler/shared/sub/sub/sharedsub.cs",
+                "src/project/compiler/resources/resource.res",
+                "src/project/compiler/resources/sub/resource2.res",
+                "src/project/compiler/resources/sub/sub/resource3.res",
+                "src/project/content1.txt",
+                "src/project/obj/object.o",
+                "src/project/bin/object",
+                "src/project2/source1.cs",
+                "src/project2/sub/source2.cs",
+                "src/project2/sub/source3.cs",
+                "src/project2/sub2/source4.cs",
+                "src/project2/sub2/source5.cs",
+                "src/project2/compiler/preprocess/preprocess-source1.cs",
+                "src/project2/compiler/preprocess/sub/preprocess-source2.cs",
+                "src/project2/compiler/preprocess/sub/sub/preprocess-source3.cs",
+                "src/project2/compiler/preprocess/sub/sub/preprocess-source3.txt",
+                "src/project2/compiler/shared/shared1.cs",
+                "src/project2/compiler/shared/shared1.txt",
+                "src/project2/compiler/shared/sub/shared2.cs",
+                "src/project2/compiler/shared/sub/shared2.txt",
+                "src/project2/compiler/shared/sub/sub/sharedsub.cs",
+                "src/project2/compiler/resources/resource.res",
+                "src/project2/compiler/resources/sub/resource2.res",
+                "src/project2/compiler/resources/sub/sub/resource3.res",
+                "src/project2/content1.txt",
+                "src/project2/obj/object.o",
+                "src/project2/bin/object",
+                "lib/source6.cs",
+                "lib/sub3/source7.cs",
+                "lib/sub4/source8.cs",
+                "res/resource1.text",
+                "res/resource2.text",
+                "res/resource3.text");
+
```

Comment:
Looks like we are reinventing some functional test utils here: https://github.com/aspnet/XRE/tree/dev/test/Microsoft.Framework.PackageManager.FunctionalTests

@davidfowl , we need to decide the DSL used to describe directory structures. Do we want the plain list above, or we want a single JSON string like:
```c#
var expectedOutputDir = @"{
  '.': ['project.json', 'Config.json', 'Program.cs', 'build_config1.bconfig'],
  'Views': {
    'Home': ['index.cshtml'],
    'Shared': ['_Layout.cshtml']
  },
  'Controllers': ['HomeController.cs'],
  'Models': ['User.cs', 'build_config2.bconfig'],
  'Build': ['build_config3.bconfig'],
  'packages': {}
}";
``` 
Plain list is straightforward but has some redundancy. JSON describes the dir tree structure in a intuitive way but might need more effort to understand.